### ISSUE TITLE
warnings building `llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp`

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#125537]
```
/home/user/CLionProjects/llvm-project/llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp:1706:6: warning: ‘bool llvm::operator==(const SDTypeConstraint&, const SDTypeConstraint&)’ has not been declared within ‘llvm’
 1706 | bool llvm::operator==(const SDTypeConstraint &LHS,
      |      ^~~~
In file included from /home/user/CLionProjects/llvm-project/llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp:14:
/home/user/CLionProjects/llvm-project/llvm/utils/TableGen/Common/CodeGenDAGPatterns.h:390:15: note: only here as a ‘friend’
  390 |   friend bool operator==(const SDTypeConstraint &LHS,
      |               ^~~~~~~~
/home/user/CLionProjects/llvm-project/llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp:1732:6: warning: ‘bool llvm::operator<(const SDTypeConstraint&, const SDTypeConstraint&)’ has not been declared within ‘llvm’
 1732 | bool llvm::operator<(const SDTypeConstraint &LHS, const SDTypeConstraint &RHS) {
      |      ^~~~
/home/user/CLionProjects/llvm-project/llvm/utils/TableGen/Common/CodeGenDAGPatterns.h:392:15: note: only here as a ‘friend’
  392 |   friend bool operator<(const SDTypeConstraint &LHS,
      |               ^~~~~~~~
```

I am using `gcc (GCC) 14.2.1 20240910` on Manjaro Linux.

**Comment by firewave** [comment]
It is strange that this warning has no type.
